server/sandbox.go

```diff
@@ -104,18 +104,19 @@ func (s *Server) RunPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 		return nil, fmt.Errorf("pod sandbox (%s) already exists", podSandboxDir)
 	}
 
-	if err = os.MkdirAll(podSandboxDir, 0755); err != nil {
-		return nil, err
-	}
-
 	defer func() {
 		if err != nil {
+			s.releasePodName(name)
 			if err2 := os.RemoveAll(podSandboxDir); err2 != nil {
 				logrus.Warnf("couldn't cleanup podSandboxDir %s: %v", podSandboxDir, err2)
 			}
 		}
 	}()
 
+	if err = os.MkdirAll(podSandboxDir, 0755); err != nil {
```

Comment:
If MkdirAll failed, path my not be created, then RemoveAll in defer will remove a path that does not exist, this won't fail. Did I get the point here, if not or this really bothers, pls enlight me